hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/SaslPropertiesResolver.java

```diff
@@ -82,7 +83,7 @@ public Configuration getConf() {
    * @return sasl Properties
    */
   public Map<String,String> getDefaultProperties() {
-    return properties;
+    return new TreeMap<>(properties);
```

Comment:
I'd propose having a private cloneProperties() method used in all calls, which includes a `requireNonNull()` on the properties...it is null until setConf() is called.

Reply:
done